src/AST.cc

```diff
@@ -3,6 +3,10 @@
 
 const Location Location::NONE(0, 0, 0, 0, nullptr);
 
+bool Location::isNone() const{
+	return (firstLine()==0 && firstColumn()==0 && lastLine()==0 && lastColumn()==0 && filePath()==nullptr);
```

Comment:
To avoid duplication: Instead of hardcoding the magic values in `NONE`, could we get those values from the `NONE` instance?
..or perhaps even implement the `==` operator?

Reply:
implementing == sounds like an awesome idea - I can imagine that we will find other uses for != and == later.